src/io/flutter/survey/FlutterSurveyNotifications.java

```diff
@@ -0,0 +1,162 @@
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+// TODO(pq): remove for M39 / post: 2019/08/24 18:00 PST.
+public class FlutterSurveyNotifications {
+  private static final int NOTIFICATION_DELAY_IN_SECS = 3;
+
+  private static final String FLUTTER_LAST_SURVEY_CHECK_KEY = "FLUTTER_LAST_SURVEY_CHECK_KEY";
+  private static final long CHECK_INTERVAL_IN_MS = TimeUnit.HOURS.toMillis(40);
+
+  private static final String SURVEY_ACTION_TEXT = "Take Flutter Q3 Survey";
+  private static final String SURVEY_TITLE = "Help improve Flutter! Take our Q3 survey.";
+
+  private static final String SURVEY_TAKEN = "io.flutter.survey.2019.q3.alreadyTaken";
+  private static final String SURVEY_URL = "https://google.qualtrics.com/jfe/form/SV_3kiGXYfYOfXUjB3?Source=IntelliJ&ClientID=";
+
+  private static long SURVEY_START;
+  private static long SURVEY_END;
+
+  static {
+    final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm");
+    dateFormat.setTimeZone(TimeZone.getTimeZone("PST"));
+    try {
+      SURVEY_START = dateFormat.parse("2019/08/12 9:00").getTime();
+      SURVEY_END = dateFormat.parse("2019/08/24 18:00").getTime();
+    }
+    catch (ParseException e) {
+      // Shouldn't happen.
+    }
+  }
+
+  private static boolean isSurveyOpen() {
+    final long now = System.currentTimeMillis();
+    return now >= SURVEY_START && now <= SURVEY_END;
+  }
+
+  interface FlutterSurveyNotifier {
+    void prompt();
+  }
+
+  public static void init(@NotNull Project project) {
+    project.getMessageBus().connect().subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, new FileEditorManagerListener() {
+      @Override
+      public void fileOpened(@NotNull final FileEditorManager source, @NotNull final VirtualFile file) {
+        check(file);
+      }
+
+      @Override
+      public void selectionChanged(@NotNull FileEditorManagerEvent event) {
+        final VirtualFile file = event.getNewFile();
+        if (file != null) {
+          check(file);
+        }
+      }
+
+      private void check(@NotNull VirtualFile file) {
+        if (PubRoot.isPubspec(file) || file.getFileType() == DartFileType.INSTANCE) {
+          new FlutterSurveyNotifications(project).checkForDisplaySurvey();
+        }
+      }
+    });
+  }
+
+  @NotNull final Project myProject;
+
+  FlutterSurveyNotifications(@NotNull Project project) {
+    this.myProject = project;
+  }
+
+  private void checkForDisplaySurvey() {
+    // Limit display to the survey window.
+    if (!isSurveyOpen()) return;
+
+    final PropertiesComponent properties = PropertiesComponent.getInstance();
+
+    // Don't prompt more often than every 40 hours.
+    final long lastCheckedMillis = properties.getOrInitLong(FLUTTER_LAST_SURVEY_CHECK_KEY, 0);
+    if (System.currentTimeMillis() - lastCheckedMillis < CHECK_INTERVAL_IN_MS) return;
+
+    properties.setValue(FLUTTER_LAST_SURVEY_CHECK_KEY, String.valueOf(System.currentTimeMillis()));
```

Comment:
I assume that `properties` values can only be strings (in terms of why you're not storing as a long)?

Reply:
👍